src/knl/knl_hdiff_variant_ij_private_blocks.cpp

```diff
@@ -0,0 +1,151 @@
+			variant_base::prerun();                                                                                                           
+            const int total_size = this->storage_size();
+            const int zero_offset = this->zero_offset();
+			value_type *__restrict__ in = this->in() - zero_offset;
+			value_type *__restrict__ coeff = this->coeff() - zero_offset;
+			value_type *__restrict__ out = this->out() - zero_offset;
+			constexpr int istride = 1;
+			const int jstride = this->jstride();
+			const int kstride = this->kstride();
+			const int isize = this->isize(); 
+			const int jsize = this->jsize(); 
+			const int ksize = this->ksize(); 
+			const int h = this->halo();
+        	double dx = 1. / (double)(isize);
+            double dy = 1. / (double)(jsize);
+            double dz = 1. / (double)(ksize);      
+
+
+#pragma omp parallel for 
+            for (int i = 0; i < total_size; ++i) {
+                int z_q = i / kstride; 
+                int z_r = i % kstride; 
+                int y_q = z_r / jstride; 
+                int y_r = z_r % jstride; 
+                int x_q = y_r; 
+                double x = dx * (double)(x_q);
+                double y = dy * (double)(y_q);
+                double z = dz * (double)(z_q);
+                in[i] = 3.0 +
+                           	1.25 * (2.5 + cos(M_PI * (18.4 * x + 20.3 * y)) +
+                            0.78 * sin(2 * M_PI * (18.4 * x + 20.3 * y) * z)) /
+                            4.;
+                coeff[i] = 1.4 +
+                      	    0.87 * (0.3 + cos(M_PI * (1.4 * x + 2.3 * y)) +
+                            1.11 * sin(2 * M_PI * (1.4 * x + 2.3 * y) * z)) /
+                            4.;
+            }
+		}
+
+		template <class ValueType>
+		void hdiff_variant_ij_private_blocks<ValueType>::hdiff() {
+			const value_type *__restrict__ in = this->in();
+			const value_type *__restrict__ coeff = this->coeff();
+			value_type *__restrict__ out = this->out();
+
+			constexpr int istride = 1;
+			const int jstride = this->jstride();
+			const int kstride = this->kstride();
+			const int h = this->halo();
+			const int isize = this->isize();
+			const int jsize = this->jsize();
+			const int ksize = this->ksize();
+
+			if (this->istride() != 1)
+				throw ERROR("this variant is only compatible with unit i-stride layout");
+			if (this->halo() < 2)
+				throw ERROR("Minimum required halo is 2");
+
+#pragma omp parallel 
+			{
+				int thread_id = omp_get_thread_num();
+				value_type *__restrict__ lap = lap_data.data() + thread_id * (m_iblocksize + 2) * (m_jblocksize + 2);
+				value_type *__restrict__ flx = flx_data.data() + thread_id * (m_iblocksize + 1) * (m_jblocksize);
+				value_type *__restrict__ fly = fly_data.data() + thread_id * (m_iblocksize) * (m_jblocksize + 1);
+#pragma omp for collapse(3)
+				for (int jb = 0; jb < jsize; jb += m_jblocksize) {
+			        for (int k = 0; k < ksize; ++k) {
```

Comment:
Indentation is off in many places, please format using clang-format if possible (there’s a `make format` target actually).

Reply:
Thanks, I hadn't noticed. I'll use clang-format